### DISCUSSION TITLE
Perfect bindings for taking screenshots in Hyprland using hyprwm/contrib grimblast

### DISCUSSION THREAD

**Discussion by txtyash** [hyprwm/Hyprland#2617] [Show and tell]
These are my bindings for taking screenshots of the fullscreen/active windows & selected regions using grim & slurp:
```
# start screenshot submap 
bind = $mainMod, s, submap, screenshot 

# submap
submap = screenshot

# fullscreen / selected area / active window
binde = , f, exec, grimblast copysave screen $HOME"/screenshots/full/"$(date +'%F-%T.png'); hyprctl dispatch submap reset
binde = , s, exec, grimblast copysave area $HOME"/screenshots/area/"$(date +'%F-%T.png'); hyprctl dispatch submap reset
binde = , a, exec, grimblast copysave active $HOME"/screenshots/active/"$(date +'%F-%T.png'); hyprctl dispatch submap reset

# reset screenshot submap
bind = , escape,submap,reset 
submap = reset
```
I use a submap. To take a screenshot of the entire screen press `mod+s` -> `f`. It will be saved under `~/screenshots/full/` with the date and the time. The screenshot also gets copied to the clipboard using `wl-copy` which I'm able to pull up anytime using `copyq`.

BTW, the submap resets automatically after taking a screenshot. To end the submap without taking a screenshot I've bind `esc`.

**Comment by MightyPlaza** [comment]
Might need some work since pressing Esc while on slurp creates an empty file

**Reply by txtyash** [reply]
Nice find! I'm not sure how this could be done for now. I'll try to figure out.

**Reply by MightyPlaza** [reply]
It outs to stderr "invalid geometry", just a simple if would work
The trouble is keeping it a command and not a script

**Reply by txtyash** [reply]
I think making it into a single installable script would be a better option then since most hyprland users would need that. I'll start writing & I'll need ur help

**Reply by MightyPlaza** [reply]
wdym like making another grimblast or something clone?

**Reply by JustSimplyKyle** [reply]
Also, you can do
```
binde = , f, exec, <grimstuff>
bind = ,f,submap,exit
```
Instead of that hyprctl "hack"

**Reply by txtyash** [reply]
Why didn't ya tell me it exists. I would've used grimblast from the beginning lol!

**Reply by MightyPlaza** [reply]
finally fixed the command
```bind = $mainMod SHIFT, S, exec, grim -g "$(slurp -w 0)" - | ifne bash -c "tee > $HOME\"/Pictures/Screenshots/Screenshot\"$(date +\"%Y-%m-%d_%H-%M- %S\")\".png\" >(wl-copy -t image/png)"```

**Comment by txtyash** [comment]
> Also, you can do
> 
> ```
> binde = , f, exec, <grimstuff>
> bind = ,f,submap,exit
> ```
> 
> Instead of that hyprctl "hack"

They'd both work just the same right?
